plugins/experimental/alt_svc/single_service_file_map.cc

```diff
@@ -0,0 +1,122 @@
+/**
+  @file
+  @brief A particular IPHostMap implementation which takes a static file and routes client IPs based on that file.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "ip_host_map.h"
+#include "prefix_parser.h"
+#include <fstream>
+#include <algorithm>
+
+using namespace std;
+
+static char const *ssfm_empty = "";
+
+string const &
+SingleServiceFileMap::findHostForIP(const sockaddr *ip, string const &hostname) const noexcept
+{
+  return this->findHostForIP(ip);
+}
+
+string const &
+SingleServiceFileMap::findHostForIP(const sockaddr *ip) const noexcept
+{
+  char *data           = nullptr;
+  string const *output = new string(this->host_map.contains(ip, (void **)&data) ? data : ssfm_empty);
```

Comment:
I was asked to change the signature to `std::string const&`, see [this conversation](https://github.com/apache/trafficserver/pull/2293#discussion_r129739016). Is there any way I can make that not do memory leak?